spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-pulsar/src/main/java/smoketest/pulsar/ReactiveAppConfig.java

```diff
@@ -14,22 +14,29 @@
  * limitations under the License.
  */
 
-package smoketest.pulsar.reactive;
+package smoketest.pulsar;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 import org.apache.pulsar.reactive.client.api.MessageSpec;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
 
 import org.springframework.boot.ApplicationRunner;
-import org.springframework.boot.SpringApplication;
-import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
 import org.springframework.pulsar.core.PulsarTopic;
+import org.springframework.pulsar.reactive.config.annotation.ReactivePulsarListener;
 import org.springframework.pulsar.reactive.core.ReactivePulsarTemplate;
 
-@SpringBootApplication
-public class SampleReactivePulsarApplication {
+@Configuration(proxyBeanMethods = false)
+@Profile("smoketest.pulsar.reactive")
```

Comment:
Using the profile approach works well w/ the `@ActiveProfiles` on the `@Nested` SBT. I guess the alternative is to use the `@SpringBootTest(classes = ...` and point directly to a specific app class (imperative or reactive).